libvmi/core.c

```diff
@@ -77,22 +77,26 @@ open_config_file(
     dbprint(VMI_DEBUG_CORE, "--looking for config file at %s\n", location);
 
     f = fopen(location, "r");
-    free(location);
 
     if (f) {
         goto success;
     }
+    free(location);
 
     /* finally check in /etc */
     dbprint(VMI_DEBUG_CORE, "--looking for config file at /etc/libvmi.conf\n");
-    f = fopen("/etc/libvmi.conf", "r");
+    location = g_malloc0(strlen("/etc/libvmi.conf")+1);
```

Comment:
AFAIK the only thing g_free does that differs from free is that it checks if the pointer is null first. So IMHO there isn't a hard requirement to use it.


On Fri Jan 16 2015 08:05:00 AM CET, Bryan Payne <notifications@github.com> wrote:

> > 
> > /* finally check in /etc */
> > dbprint(VMI_DEBUG_CORE, "--looking for config file at
> > /etc/libvmi.conf\n"); -       f = fopen("/etc/libvmi.conf", "r");
> > +       location = g_malloc0(strlen("/etc/libvmi.conf")+1);
> 
> If you use g_malloc0 then you need to use g_free to cleanup that memory
> later (not free).   I see you have used free a couple of times below.
> 
> For the most part, we use safe_malloc in libvmi (a macro unique to this
> project).   However, I just realized that several g_malloc0 calls have
> slipped in as well.   And those other g_malloc0's appear to be using the
> wrong free.   Alas, I can clean those up shortly.
> 
> For the purposes of this PR, it would be nice to just stick with
> safe_malloc and then the calls to free are all ok as they are.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/libvmi/libvmi/pull/231/files#r23065754